tl/candidate_generation/utility.py

```diff
@@ -100,6 +112,13 @@ def create_cfd_canonical(self, df, relevant_columns, column, size, properties, m
                     cf_dict['pagerank'] = candidate_dict[kg_id]['pagerank_float']
                     cf_dict[self.score_column_name] = candidate_dict[kg_id]['score']
                     candidates_format.append(cf_dict)
+        
+        max_threads = 50
```

Comment:
This should be:
1. A parameter with a default value of 50
2. min(number of rows in the file, 50)